osu.Game.Rulesets.Osu/Skinning/Legacy/OsuLegacySkinTransformer.cs

```diff
@@ -163,7 +163,10 @@ public OsuLegacySkinTransformer(ISkin skin)
                         return null;
 
                     case OsuSkinComponents.ApproachCircle:
-                        return new LegacyApproachCircle();
+                        if (IsProvidingLegacyResources)
```

Comment:
Are you sure this matches stable? Should it not check specifically for the `approachcircle` texture?

Reply:
This technically won't have any difference because `LegacyApproachCircle` is actually a skinnable drawable that will still look up for the nearest `approachcircle` texture it finds in the skin source chain (as it has been in master). Since you're already confused about it, I'll go ahead and refactor it to give it some sense.